src/Components/AllArticles.jsx

```diff
@@ -1,22 +1,39 @@
 import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
 import getAllArticles from "../../utils/get-all-articles";
 
 function AllArticles() {
   const [allArticles, setAllArticles] = useState([]);
+  const navigate = useNavigate();
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     getAllArticles().then((articles) => {
+      setIsLoading(false);
       setAllArticles(articles);
     });
   }, []);
 
+  if (isLoading) {
+    return <p>loading...</p>;
+  }
+
+  const handleClick = (articleId) => {
+    navigate(`/articles/${articleId}`);
```

Comment:
While this works, it is not a method that is accessible to screen readers. This will not tell a partially sighted user that they can click it. Make sure to look into aria roles if this is the method you wish to stick with.